src/client.rs

```diff
@@ -801,9 +801,17 @@ impl AudioHandler {
         let mut config: StreamConfig = config.into();
         config.channels = format0.channels as _;
         match sample_format {
-            cpal::SampleFormat::F32 => self.build_output_stream::<f32>(&config, &device)?,
+            cpal::SampleFormat::I8 => self.build_output_stream::<i8>(&config, &device)?,
             cpal::SampleFormat::I16 => self.build_output_stream::<i16>(&config, &device)?,
+            cpal::SampleFormat::I32 => self.build_output_stream::<i32>(&config, &device)?,
+            cpal::SampleFormat::I64 => self.build_output_stream::<i64>(&config, &device)?,
+            cpal::SampleFormat::U8 => self.build_output_stream::<u8>(&config, &device)?,
             cpal::SampleFormat::U16 => self.build_output_stream::<u16>(&config, &device)?,
+            cpal::SampleFormat::U32 => self.build_output_stream::<u32>(&config, &device)?,
+            cpal::SampleFormat::U64 => self.build_output_stream::<u64>(&config, &device)?,
+            cpal::SampleFormat::F32 => self.build_output_stream::<f32>(&config, &device)?,
+            cpal::SampleFormat::F64 => self.build_output_stream::<f64>(&config, &device)?,
+            f => bail!("unsupported audio format: {:?}", f),
```

Comment:
can remove this f to support all format?

Reply:
and how about more than 2 channels mentioned in the issue. we can convert to 2 channels.

Reply:
can't remove f